Lib/fontTools/ttLib/ttFont.py

```diff
@@ -684,16 +689,28 @@ def getGlyphSet(self, preferCFF=True):
 		If the font contains both a 'CFF '/'CFF2' and a 'glyf' table, you can use
 		the 'preferCFF' argument to specify which one should be taken. If the
 		font contains both a 'CFF ' and a 'CFF2' table, the latter is taken.
+
+		If the 'location' parameter is set, it should be a dictionary mapping
+		four-letter variation tags to their float values, and the returned
+		glyph-set will represent an instance of a variable font at that location.
+		If the 'normalized' variable is set to True, that location is interpretted
+		as in the normalized (-1..+1) space, otherwise it is in the font's defined
+		axes space.
 		"""
 		glyphs = None
 		if (preferCFF and any(tb in self for tb in ["CFF ", "CFF2"]) or
 		   ("glyf" not in self and any(tb in self for tb in ["CFF ", "CFF2"]))):
 			table_tag = "CFF2" if "CFF2" in self else "CFF "
+			if location:
+				raise NotImplementedError # TODO
 			glyphs = _TTGlyphSet(self,
 			    list(self[table_tag].cff.values())[0].CharStrings, _TTGlyphCFF)
 
 		if glyphs is None and "glyf" in self:
-			glyphs = _TTGlyphSet(self, self["glyf"], _TTGlyphGlyf)
+			if location and 'gvar' in self:
+				glyphs = _TTVarGlyphSet(self, location=location, normalized=normalized)
+			else:
```

Comment:
I think if `location is not None and "gvar" not in self` (i.e. non variable font) we should error out instead of silently return the default glyph set

Reply:
> I think if `location is not None and "gvar" not in self` (i.e. non variable font) we should error out instead of silently return the default glyph set

That's debatable.  What if location points to an axis that doesn't exist in the varfont then? That's the same as location and no varfont... So I'm leaning towards the code as is.